app/scripts/controllers/reports/RunReportsController.js

```diff
@@ -328,13 +328,26 @@
                             scope.hideTable = true;
                             scope.hidePentahoReport = false;
                             scope.hideChart = true;
-                            scope.baseURL = $rootScope.hostUrl + API_VERSION + "/runreports/" + encodeURIComponent(scope.reportName);
-                            scope.baseURL += "?output-type=" + encodeURIComponent(scope.formData.outputType) + "&tenantIdentifier=" + $rootScope.tenantIdentifier + "&locale=" + scope.optlang.code + "&dateFormat=" + scope.df;
+
+                            var reportURL = $rootScope.hostUrl + API_VERSION + "/runreports/" + encodeURIComponent(scope.reportName);
+                            reportURL += "?output-type=" + encodeURIComponent(scope.formData.outputType) + "&tenantIdentifier=" + $rootScope.tenantIdentifier + "&locale=" + scope.optlang.code + "&dateFormat=" + scope.df;
 
                             var inQueryParameters = buildReportParms();
-                            if (inQueryParameters > "") scope.baseURL += "&" + inQueryParameters;
-                            // allow untrusted urls for iframe http://docs.angularjs.org/error/$sce/insecurl
-                            scope.baseURL = $sce.trustAsResourceUrl(scope.baseURL);
+                            if (inQueryParameters > "") reportURL += "&" + inQueryParameters;
+
+                            // allow untrusted urls for ajax request 
+                            // http://docs.angularjs.org/error/$sce/insecurl
+                            reportURL = $sce.trustAsResourceUrl(reportURL);
+
+                            http.get(reportURL).
+                              success(function(data, status, headers, config) {
+                                // Pass the form data to the iframe as a data url.
+                                scope.baseURL = $sce.trustAsResourceUrl('data:text/html,' + data);
```

Comment:
Thanks @lancegood. While this would work for HTML outputs, they would break PDF / Excel outputs...